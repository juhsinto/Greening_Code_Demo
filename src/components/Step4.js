import React from "react";

import {useDispatch, useSelector} from "react-redux";
import { updatePage1, updatePage2, updatePage3, updateFormProgress } from "../rootSlice";
import { useHistory } from "react-router-dom";

import { Link } from "react-router-dom";
import {Result} from "./Result"
import {Header} from "./Header"

import {Jumbotron,Container,Button} from 'react-bootstrap';


export const Step4 = () => {

    const dispatch = useDispatch();
    const history = useHistory();

    const reset = () => {
        dispatch(
            updatePage1({})
        );
        dispatch(
            updatePage2({})
        );
        dispatch(
            updatePage3({})
        );
        dispatch(updateFormProgress(0));
        history.push("./");
    };

    return (
        <Container className="p-3">
            <Jumbotron>
                <Header message="Result Page"/>
                    
                <Result />
                <br />
                <Link to="/">Start Over</Link>
                <br />
                <Button variant="danger" onClick={reset}>Reset</Button>
            </Jumbotron>
            
        </Container>
    )
}

export default Step4;
