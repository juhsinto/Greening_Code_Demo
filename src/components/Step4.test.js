import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from "react-router-dom";

import Step4 from './Step4';
import { store } from '../store';

it('renders Step4 correctly', () => {
  const tree = renderer.create(<Router><Provider store={store}><Step4 /></Provider></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});