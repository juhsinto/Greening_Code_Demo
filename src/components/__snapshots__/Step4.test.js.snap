// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Step4 correctly 1`] = `
<div
  className="p-3 container"
>
  <div
    className="jumbotron"
  >
    <h1
      className="header"
    >
      Result Page
    </h1>
    <p>
      JSON data:
    </p>
    <pre>
      {
  "page1": {},
  "page2": {},
  "page3": {},
  "formProgress": 0
}
    </pre>
    <p>
      Form Progress:
    </p>
    <div
      className="progress"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        0%
      </div>
    </div>
    <br />
    <a
      href="/"
      onClick={[Function]}
    >
      Start Over
    </a>
    <br />
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
